pkg/util/indexer/indexer.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package indexer
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1alpha2"
+)
+
+var (
+	IndexQueueClusterQueue = func(obj client.Object) []string {
```

Comment:
These index functions both used in package `cache` and `queue`, so moved to `util/indexer`.

Reply:
Maybe we should just have a single `SetupIndexes` function for the kueue APIs.

Reply:
Do you mean have only one `setupIndexes()` in main.go? Not separated into different packages like `queue`, `cache`, `job`.

Reply:
probably not in main.go, because the tests need it. But yes, a single place.

Reply:
Let me make this cleanup tomorrow. We can leave this a follow up if that's ok with you.

Reply:
sg